### PR TITLE
Restore compilation of AdaEurope2018 examples

### PR DESCRIPTION
**Pull request by anisimkov**

### CODE REVIEW

File: contrib/AdaEurope2018/rewrite_self_arg.adb [hunk 1]
```diff
@@ -4,18 +4,19 @@
 with Ada.Text_IO;
 
 with Langkit_Support.Text, Libadalang.Analysis, Libadalang.Iterators,
-     Libadalang.Rewriting;
+     Libadalang.Rewriting, Libadalang.Common;
```
Comment (pmderodat) [inline contrib/AdaEurope2018/rewrite_self_arg.adb:7]: Please keep the packages sorted (by name)

File: contrib/AdaEurope2018/derivation_count.adb [hunk 2]
```diff
@@ -5,17 +5,19 @@ with Ada.Containers.Hashed_Maps;
...
+with Libadalang.Common;
 
 with Helpers;
 
 procedure Derivation_Count is
 
    package TIO renames Ada.Text_IO;
    package LAL renames Libadalang.Analysis;
-   use type LAL.Ada_Node_Kind_Type;
+   package LAC renames Libadalang.Common;
```
Comment (pmderodat) [inline contrib/AdaEurope2018/derivation_count.adb:16]: Could you rename this to `LALCO`, for consistency with the rest of the codebase?

**Review by pmderodat — CHANGES_REQUESTED** [review]
Thanks for taking the time to do this! LGTM after minor changes.

File: contrib/AdaEurope2018/rewrite_self_arg.adb (hunk 1 above)
Comment (anisimkov) [reply contrib/AdaEurope2018/rewrite_self_arg.adb]: Done

File: contrib/AdaEurope2018/derivation_count.adb (hunk 2 above)
Comment (anisimkov) [reply contrib/AdaEurope2018/derivation_count.adb]: Done

File: contrib/AdaEurope2018/derivation_count.adb [hunk 3]
```diff
@@ -43,15 +46,15 @@ procedure Derivation_Count is
...
-      if Node.Kind /= LAL.Ada_Derived_Type_Def then
-         return LAL.Into;
+      if Node.Kind /= LALCO.Ada_Derived_Type_Def then
+         return LALCO.Into;
       end if;
 
       Derived := Node.As_Derived_Type_Def;
       if Derived.F_Has_With_Private.P_As_Bool
-        or else not Derived.P_Is_Tagged_Type
+        or else not Derived.As_Base_Type_Decl.P_Is_Tagged_Type
```
Comment (pmderodat) [inline contrib/AdaEurope2018/derivation_count.adb:55]: This conversion will always fail: `Base_Type_Decl` does not derive from `Derived_Type_Def` and conversely. However I think that all `Derived_Type_Def` nodes have a `Type_Decl` parent node, so this should become `Derived.Parent.As_Type_Decl`.

@Roldak could you confirm?

Comment (anisimkov) [reply contrib/AdaEurope2018/derivation_count.adb]: I just tried both variants to run, and @pmderodat new variant is working when prev variant

Comment (pmderodat) [reply contrib/AdaEurope2018/derivation_count.adb]: Great, thanks!

**Review by pmderodat — APPROVED** [review]

**Pull request merged**